src/bindings/cpp/onion.hpp

```diff
@@ -26,6 +26,7 @@
 
 #include <string>
 #include <onion/onion.h>
+#include "listen_point.hpp"
```

Comment:
When I remove the include and replace it with a forward declaration, I get an error with the build. I believe that the full include is needed because in the addListenPoint method, I tried to keep the code inline and it needs access to the c_handler method. I don't think there is anyway around it.